src/rp2_common/pico_flash/flash.c

```diff
@@ -142,16 +142,14 @@ static int default_enter_safe_zone_timeout_ms(__unused uint32_t timeout_ms) {
         // it only prevents the other core from also entering a critical section.
         // Therefore, we must do our own handshake which starts a task on the other core and have it disable interrupts
         uint core_num = get_core_num();
-        // create at low priority
+        // create at low priority on other core
         TaskHandle_t task_handle;
-        if (pdPASS != xTaskCreate(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, &task_handle)) {
+        if (pdPASS != xTaskCreateAffinitySet(flash_lockout_task, "flash lockout", configMINIMAL_STACK_SIZE, (void *)core_num, 0, 1u << (core_num ^ 1), &task_handle)) {
             return PICO_ERROR_INSUFFICIENT_RESOURCES;
         }
         lockout_state[core_num] = FREERTOS_LOCKOUT_LOCKER_WAITING;
```

Comment:
I wonder if this line should be moved before the call to xTaskCreateAffinitySet

Reply:
Hmm yes it probably should, as otherwise it'll use the previous lockout state.

Reply:
and then there's no point creating it with a low priority? May as well make it super high immediately?

Reply:
All sounds reasonable! I'll change it to do that and give it a test tomorrow :+1:

Reply:
Ignore my comments. On reflection, I prefer it the way it is.